frontend/pages/combo-finder.vue

```diff
@@ -0,0 +1,251 @@
+import Vue from "vue";
+import ArtCircle from "@/components/ArtCircle.vue";
+import ComboResults from "@/components/search/ComboResults.vue";
+import {
+  findCombosFromDecklist,
+  convertDecklistToDeck,
+} from "@/lib/decklist-parser";
+
+import type { FormattedApiResponse } from "@/lib/api/types";
+import type Card from "@/lib/api/models/card";
+
+type ComboFinderData = {
+  decklist: string;
+  numberOfCardsInDeck: number;
+  lookupInProgress: boolean;
+  combosInDeck: FormattedApiResponse[];
+  potentialCombos: FormattedApiResponse[];
+  missingDecklistCards: Card[];
+};
+
+const LOCAL_STORAGE_DECK_STORAGE_KEY =
+  "commander-spellbook-combo-finder-last-decklist";
+
+export default Vue.extend({
+  components: {
+    ArtCircle,
+    ComboResults,
+  },
+  data(): ComboFinderData {
+    return {
+      decklist: "",
+      numberOfCardsInDeck: 0,
+      lookupInProgress: false,
+      combosInDeck: [],
+      potentialCombos: [],
+      missingDecklistCards: [],
+    };
+  },
+  computed: {
+    numberOfCardsText(): string {
+      return `${this.numberOfCardsInDeck} ${this.$pluralize(
+        "card",
+        this.numberOfCardsInDeck
+      )}`;
+    },
+    combosInDeckHeadingText(): string {
+      const numOfCombos = this.combosInDeck.length;
+
+      if (numOfCombos === 0) {
+        return "No Combos Found";
+      }
+
+      return `${numOfCombos} ${this.$pluralize("Combo", numOfCombos)} Found`;
+    },
+    potentialCombosInDeckHeadingText(): string {
+      const numOfCombos = this.potentialCombos.length;
+
+      return `${numOfCombos} Potential ${this.$pluralize(
+        "Combo",
+        numOfCombos
+      )} Found`;
+    },
+  },
+  mounted() {
+    const savedDeck = localStorage.getItem(LOCAL_STORAGE_DECK_STORAGE_KEY);
+
+    if (savedDeck?.trim()) {
+      this.decklist = savedDeck;
+      this.lookupCombos();
+    }
+  },
+  methods: {
+    async lookupCombos() {
+      const deck = convertDecklistToDeck(this.decklist);
+      this.numberOfCardsInDeck = deck.numberOfCards;
+
+      this.combosInDeck = [];
+      this.potentialCombos = [];
+      this.missingDecklistCards = [];
+      this.lookupInProgress = true;
+
+      // not possible to have any combos if deck has 1
+      // or fewer card in it, so we can skip the lookup
+      if (this.numberOfCardsInDeck < 2) {
+        this.lookupInProgress = false;
+        return;
+      }
+
+      localStorage.setItem(LOCAL_STORAGE_DECK_STORAGE_KEY, this.decklist);
+
+      // TODO: let's debounce in case some weirdo is manually
+      // typing in their deck list
+      const combos = await findCombosFromDecklist(deck.cards);
+
+      this.combosInDeck = combos.combosInDecklist;
+      this.potentialCombos = combos.potentialCombos;
+      this.missingDecklistCards = combos.missingCardsForPotentialCombos;
+
+      this.lookupInProgress = false;
+    },
+
+    clearDecklist() {
+      this.decklist = "";
+      this.combosInDeck = [];
+      this.potentialCombos = [];
```

Comment:
Do we really need new arrays or just empty arrays?

```suggestion
      this.combosInDeck.length = 0;
      this.potentialCombos.length = 0;
```

Reply:
Yeah, looks like setting the `length` property is still a problem in Vue 2: https://v2.vuejs.org/v2/guide/reactivity.html#For-Arrays

Reply:
Though `this.combosInDeck.splice(0)` could be used instead.